Updated SSD1306 OLED Library's URL

styropyr0 opened the pull request:
I have my library's URL changed from https://github.com/styropyr0/oled.h to https://github.com/styropyr0/SSD1306. Please approve the changes or if there are things that need correction, please let me know.

styropyr0 commented:
@ArduinoBot

per1234 reviewed (commented):
> I have my library's URL changed from https://github.com/styropyr0/oled.h to https://github.com/styropyr0/SSD1306.

This is false. It seems to be the very opposite. If I open this URL in my browser:

https://github.com/styropyr0/SSD1306

It is redirected to this URL:

https://github.com/styropyr0/oled.h 

GitHub automatically sets up a redirect when you change a repository's name, so this tells me that you changed the repository name from "SSD1306" to "oled.h".

The bot resolves the URL from your pull request before checking for duplicates. So even though you used the URL `https://github.com/styropyr0/SSD1306` in your pull request, the bot checks for existing registrations of the true URL `https://github.com/styropyr0/oled.h`. So this is why it returned the error:

> ❌ ERROR: Submission URL is already in the Library Manager index.

If you want to rename your repository, that is fine and we are happy to update the registration for the library. But you need to actually rename the repository before you submit the URL change request here.

@styropyr0 in the future please only create one PR for each request.

You will not accomplish anything by submitting the pull request multiple times. That will only waste your time and the time of the humans who maintain this free open source project.

If you have any questions, concerns, or problems, you are welcome to write a clear and detailed request in a comment here. I'll be happy to provide assistance.

**⚠ Repeated irresponsible behavior may result in the revocation of your Library Manager Registry privileges.**

Thanks in advance for your cooperation.

---

If you have any questions, concerns, or problems, you are welcome to write a clear and detailed description in a comment on the pull request thread. I'll be happy to provide assistance.

styropyr0 commented:
Sorry for making things complicated. I have updated my repo's url to the one I was trying to add but the linter repeatedly gave me errors after fixing the following issue-

1. Change the example file's name from Fragments 1.ino to FragmentsAdvanced.ino.

This issue was resolved and updated in the main branch, but it was still raising the error. At that point, I just needed more time to look into the issue. I wasn't sure about what has went wrong. Since, it was raising error even after the fix, i closed the PR and reverted the url back to the original one.

per1234 commented:
Thanks for the clarification.

>This issue was resolved and updated in the main branch, but it was still raising the error.

Library Manager works exclusively from Git tags, so the bot will only see fixes you make after you create a new [release](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository) or [tag](https://git-scm.com/docs/git-tag) in the library repository.

So my hypothesis is that, at the time the error was being raised, the fix was not present in the latest tagged point in the repository's revision history.

From a quick look, it appears the problem is fixed in the `v1.0.3` tag, so I think everything is good now.

styropyr0 commented:
That makes sense.  Thank you!
